src/pages/LoginToAdminPage.js

```diff
@@ -36,40 +36,57 @@ function LoginToAdminPage() {
         setPlayersList(tempArray);
     }
     
+    const getSuccessfullFechedData = async (password) => {
+        setPassword(password)
+        const result = await fetch(
+            `${apiUrl}/admin/players/`,
+            {   
+                headers: { 'Authorization': `Token ${password}`}
+            });
+        let responseData = await result.json();
+        if (result.status === 200) {
+            responseData = responseData.map(item => {
+                return {
+                    ...item,
+                    selected: false
+                };
+            })
+            setPlayersList(responseData);
+            setAuthorized(true);
+        };
+        if (result.status === 403) {
+            setShowPasswordErrorMessage(true)
+        };
+    }
+
     const getPlayersDataHandler = async (password) => {
         const adminPassword = password
         console.log('adminPassword', adminPassword)
-        setPassword(adminPassword)
 
         setIscallLoading(true);
         setShowPasswordErrorMessage(false);
 
         try {
-            const result = await fetch(
-                `${apiUrl}/admin/players/`,
-                {   
-                    headers: { 'Authorization': `Token ${adminPassword}`}
-                });
-            let responseData = await result.json();
-            if (result.status === 200) {
-                responseData = responseData.map(item => {
-                    return {
-                        ...item,
-                        selected: false
-                    };
-                })
-                setPlayersList(responseData);
-                setAuthorized(true);
-            };
-            if (result.status === 403) {
-                setShowPasswordErrorMessage(true)
-            };
+            getSuccessfullFechedData(adminPassword)
         } catch (error) {
             console.log(error);
             window.location = "/500"
         };
         setIscallLoading(false);
     }
+    // UseEffect content will disply in the first place one time.
+    // if some state is inside of array [] and updated, useEffect will be called again.  
+    useEffect(() => {
+        const storedPassword = sessionStorage.getItem('storedPassword')
+        console.log('>>>',storedPassword)
+        if( storedPassword ) {
+            console.log('>>>effect called one time');
+            getSuccessfullFechedData(storedPassword)
```

Comment:
When a function that you want to use contains API calls or an external library uses, it's always a good idea to surround the code with `try/catch` block. Let's put `try/catch` here too.